src/controller/event/handler/internal/artifact.go

```diff
@@ -321,8 +324,14 @@ func (a *ArtifactEventHandler) onDelete(ctx context.Context, event *event.Artifa
 		log.Errorf("failed to delete scan reports of artifact %v, error: %v", unrefDigests, err)
 	}
 
-	if event.Artifact.Type == sbom.ArtifactTypeSBOM && len(event.Artifact.Digest) > 0 {
-		if err := reportMgr.DeleteByExtraAttr(ctx, v1.MimeTypeSBOMReport, "sbom_digest", event.Artifact.Digest); err != nil {
+	// delete sbom_report when the subject artifact is deleted
+	if err := sbom.Mgr.DeleteByArtifactID(ctx, event.Artifact.ID); err != nil {
+		log.Errorf("failed to delete scan reports of artifact ID %v, error: %v", event.Artifact.ID, err)
```

Comment:
maybe change the log message to:
```
log.Errorf("failed to delete scan reports of sbom of artifact ID %v, error: %v", event.Artifact.ID, err)
```
to emphasize deleting scan reports of `sbom`.

Reply:
Updated